watertap/property_models/ion_DSPMDE_prop_pack.py

```diff
@@ -0,0 +1,814 @@
+                                       block will deal with fixing/unfixing.
+            solver : Solver object to use during initialization if None is provided
+                     it will use the default solver for IDAES (default = None)
+            hold_state : flag indicating whether the initialization routine
+                         should unfix any state variables fixed during
+                         initialization (default=False).
+                         - True - states variables are not unfixed, and
+                                 a dict of returned containing flags for
+                                 which states were fixed during
+                                 initialization.
+                        - False - state variables are unfixed after
+                                 initialization by calling the
+                                 release_state method
+        Returns:
+            If hold_states is True, returns a dict containing flags for
+            which states were fixed during initialization.
+        """
+        # Get loggers
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="properties")
+
+        # Set solver and options
+        opt = get_solver(solver, optarg)
+
+        # Fix state variables
+        flags = fix_state_vars(self, state_args)
+        # Check when the state vars are fixed already result in dof 0
+        for k in self.keys():
+            dof = degrees_of_freedom(self[k])
+            if dof != 0:
+                raise InitializationError("\nWhile initializing {sb_name}, the degrees of freedom "
+                                           "are {dof}, when zero is required. \nInitialization assumes "
+                                           "that the state variables should be fixed and that no other "
+                                           "variables are fixed. \nIf other properties have a "
+                                           "predetermined value, use the calculate_state method "
+                                           "before using initialize to determine the values for "
+                                           "the state variables and avoid fixing the property variables."
+                                           "".format(sb_name=self.name, dof=dof))
+
+        # ---------------------------------------------------------------------
+        skip_solve = True  # skip solve if only state variables are present
+        for k in self.keys():
+            if number_unfixed_variables(self[k]) != 0:
+                skip_solve = False
+
+        if not skip_solve:
+            # Initialize properties
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                results = solve_indexed_blocks(opt, [self], tee=slc.tee)
+                if not assert_optimal_termination:
```

Comment:
Two issues here:

1. This needs an argument of the `results` object.
2. You should use `check_optimal_convergence` instead of `assert_optimal_convergence` (otherwise you will get a Pyomo error here instead of the following `InitializationError`.

So: `if not check_optimal_convergence(results)`

Reply:
@andrewlee94 - Oops. I just committed the fix.